### DISCUSSION TITLE
shrink_{left,right,up,down}

### DISCUSSION THREAD

**Discussion by nazar-pc** [qtile/qtile#3782] [Ideas]
With columns layout specifically when having more than 2 columns/rows behavior is inconsistent with 2 columns/rows.

If there are 2 columns, calling `grow_left` on left column causes column to shrink. This feels natural until you add a third column, then middle column will grow with both `grow_left` and `grow_right`, which makes sense if you think about it, but then question arises whether shrinking should also be supported. With 4+ columns lack of shrinking hurts quite a bit.

**Comment by SweepingBishops** [comment]
I have experienced a few personal problems with columns as well. I now use plasma as my main layout. 
https://github.com/numirias/qtile-plasma

**Reply by nazar-pc** [reply]
I tried it before, but it wasn't updated for over a year. Does it still work fine with latest Qtile version?

**Reply by elParaguayo** [reply]
It should do. We've not really touched anything that would cause layouts to break.

I do wonder whether it's worth incorporating this layout into the main codebase.

**Reply by nazar-pc** [reply]
I'll give it is try again.

Upstreaming plasma would be great, it is probably the most advanced Qtile layout out there by far.

**Reply by elParaguayo** [reply]
That's also the problem. There's a lot of additional code to maintain. 

I do wonder if there's an easier way to recreate it.

**Reply by SweepingBishops** [reply]
> I tried it before, but it wasn't updated for over a year. Does it still work fine with latest Qtile version?

It works just fine. 

> That's also the problem. There's a lot of additional code to maintain.
> 
> I do wonder if there's an easier way to recreate it.

Plasma provides the functionalities of a few other layouts, columns stack etc. Maybe remove them (or not, as they do have functionalities that plasma doesn't have).
